server/controllers/events.coffee

```diff
@@ -50,20 +50,29 @@ module.exports.create = (req, res) ->
             if data.import
                 res.status(201).send event
             else
-                # We first share the events, if the event is shared (this
-                # condition is checked within `sendShareInvitations`)
-                ShareHandler.sendShareInvitations event, (err, updatedEvent) ->
+                # If at least one guest has the event shared
+                if req.query.share is 'true'
```

Comment:
What about putting the client side code above right here ? Without the `method` test, as :
```coffeescript
attendeeToShareWith = attendees.find (attendee) ->
    return guest.shareWithCozy
if attendeeToShareWith?
    # [...] 
```